eclair-core/src/main/scala/fr/acinq/eclair/channel/Commitments.scala

```diff
@@ -144,9 +144,17 @@ case class Commitments(channelVersion: ChannelVersion,
     localCommit.spec.htlcs.collect(incoming).filter(nearlyExpired)
   }
 
-  def addLocalProposal(proposal: UpdateMessage): Commitments = Commitments.addLocalProposal(this, proposal)
-
-  def addRemoteProposal(proposal: UpdateMessage): Commitments = Commitments.addRemoteProposal(this, proposal)
+  /**
+   * Return a fully signed commit tx, that can be published as-is.
+   */
+  def fullySignedLocalCommitTx(keyManager: ChannelKeyManager): CommitTx = {
+    val unsignedCommitTx = localCommit.publishableTxs.commitTx
+    val localSig = keyManager.sign(unsignedCommitTx, keyManager.fundingPublicKey(localParams.fundingKeyPath), TxOwner.Local, commitmentFormat)
+    val remoteSig = localCommit.publishableTxs.remoteSig
+    val commitTx = Transactions.addSigs(unsignedCommitTx, keyManager.fundingPublicKey(localParams.fundingKeyPath).publicKey, remoteParams.fundingPubKey, localSig, remoteSig)
+    Transactions.checkSpendable(commitTx).get
```

Comment:
Are we using this `.get` as some kind of `require`? We should have verified that the signed commit tx was ok when receiving their `commit_sig`, so it's here only to catch developer mistakes, right?

What about:
```scala
// We verify the remote signature when receiving their commit_sig, so this check should always pass.
require(Transactions.checkSpendable(commitTx).isSuccess, "commit signatures are invalid")
```

Reply:
It's actually a left-over from my testing, I think I should remove it.

Reply:
What about keeping it in a `require` to ensure we catch potential issues?

Reply:
Done with 2adf25e0e.